Pod/Classes/ViewControllers/ARComponentViewController.m

```diff
@@ -125,4 +130,38 @@ - (void)setProperty:(id)value forKey:(NSString *)key
     }
 }
 
+- (void)updateSafeAreaInsets
+{
+    if (self.safeAreaInsetsAreSet) {
+        return;
+    }
+
+    UIEdgeInsets safeAreaInsets = UIEdgeInsetsZero;
+    if (@available(iOS 11.0, *)) {
+        safeAreaInsets = self.view.safeAreaInsets;
+
+        // Once we receive the correct top inset value it gets resetted to 0 after the VC gets hidden
+        // So let's not reset it afterwards
+        if (self.view.safeAreaInsets.top > 0) {
+            self.safeAreaInsetsAreSet = YES;
+        }
+    } else {
+        self.safeAreaInsetsAreSet = YES;
```

Comment:
This `if` statement makes sense, but let's make sure to test both Emission and Eigen in pre-iOS 11 simulators and devices. I have some here we can use 👍

Reply:
good thinking! I just checked iOS 10 and it was broken 👀 

Seems the `safeAreaInsets` prop is not being set on iOS 10 but I don't see an obvious reason why, and I have to sign off now. Any ideas? If not I'll debug tomorrow morning.

Reply:
Oh weird – I would expect `UIEdgeInsetsZero` to get passed through in iOS 10. My only idea would be that the `viewSafeAreaInsetsDidChange` function won't get called, so we're relying on `viewDidLoad` alone, which might have a different runtime behaviour than `viewSafeAreaInsetsDidChange`. I'll be online Friday too, if you want to debug together 👍

Reply:
OK this is fixed, and I simplified the injection code significantly too.

As far as I could tell, `setProperty` wasn't triggering an update on iOS 10, so I moved the pre-ios-11 code to the init method.